cli/src/commit_templater.rs

```diff
@@ -621,6 +666,22 @@ fn builtin_commit_methods<'repo>() -> CommitTemplateBuildMethodFnMap<'repo, Comm
             Ok(L::wrap_boolean(out_property))
         },
     );
+    map.insert(
+        "signature",
+        |language, _diagnostics, _build_ctx, self_property, function| {
+            function.expect_no_arguments()?;
+            Ok(
+                language.wrap_cryptographic_signature_opt(TemplateFunction::new(
+                    self_property,
+                    |commit| {
+                        Ok(commit
+                            .is_signed()
+                            .then(|| CryptographicSignature::new(commit)))
+                    },
+                )),
+            )
+        },
```

Comment:
`self_property.map()` can be used here
```rust
            let out_property = self_property.map(|commit| {
                commit
                    .is_signed()
                    .then_some(CryptographicSignature::new(commit))
            });
            Ok(L::wrap_cryptographic_signature_opt(out_property))
```

Reply:
Addressed.
However we can't use `then_some()`, because it evaluates eagerly and we [panic in `CryptographicSignature::new()`](https://github.com/pylbrecht/jj/blob/1d3c3b8ab207e66c978ed026a7393682989111c5/cli/src/commit_templater.rs#L1740-L1742) for unsigned commits. I left a comment in the code.

Let me know if that works for you.